Data/Text.hs

```diff
@@ -7,6 +7,10 @@
 {-# LANGUAGE TypeFamilies #-}
 #endif
 
+#if defined(__GLASGOW_HASKELL__) && !defined(__HADDOCK__)
```

Comment:
Sorry for the late reply - this was copied from elsewhere in the code, I should probably have a look at how this handles haddock at some point too, but there's plenty more other improvements that need to be made (i.e., I don't think this code should live in the module it is, so this will be moved elsewhere too).